testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/deployment/JndiViewOperationJMSTestCase.java

```diff
@@ -0,0 +1,96 @@
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.integration.jca.deployment;
+
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.test.integration.jca.JcaMgmtBase;
+import org.jboss.as.test.integration.jca.rar.MultipleAdminObject1;
+import org.jboss.as.test.integration.management.util.MgmtOperationException;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test the situation when connection factory or JMS destination is configured for resource adapter
+ *
+ * @author <a href="parsharma@redhat.com">Parul Sharma</a>
+ */
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class JndiViewOperationJMSTestCase extends JcaMgmtBase {
+
+    static final String rarDeploymentName = "eis.rar";
+
+    @Deployment(name = "rar", order = 1)
+    public static Archive<?> deploytRar() {
+        ResourceAdapterArchive raa = ShrinkWrap.create(ResourceAdapterArchive.class, rarDeploymentName);
+
+        JavaArchive ja = ShrinkWrap.create(JavaArchive.class, "lib.jar");
+        ja.addPackage(MultipleAdminObject1.class.getPackage());
+        raa.addAsLibrary(ja);
+
+        raa.addAsManifestResource(JndiViewOperationJMSTestCase.class.getPackage(), "ra.xml", "ra.xml");
+        return raa;
+    }
+
+
+     @Test
+     public void testValueOfJms() throws Exception{
+        ModelNode address = getAddress();
+        ModelNode operation = getOperation(address);
+          try{
+             ModelNode result=executeOperation(operation);
+             ModelNode value = result.get("java: contexts", "java:", "JmsXA","value");
+             Boolean val = !(value.equals(null));
+             Assert.assertTrue(val);
```

Comment:
I suggest some changes here:
1.  If `result` is `undefined`, then `val` will be `true` and the test will pass, getting a false negative.
2. `value.equals(null)` never will be true, because in that case you will get a NPE.
3. `Boolean val`, you can use a primitive and avoid auto-boxing/unboxing.